internal/resources/app_endpoint_oidc_default.go

```diff
@@ -0,0 +1,274 @@
+	resp.TypeName = req.ProviderTypeName + "_app_endpoint_default_oidc_provider"
+}
+
+func (r *AppEndpointDefaultOidcProvider) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = AppEndpointDefaultOidcProviderSchema()
+}
+
+func (r *AppEndpointDefaultOidcProvider) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.Data = data
+}
+
+// ImportState imports a resource into Terraform state.
+func (r *AppEndpointDefaultOidcProvider) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// The import ID is the app endpoint name to align with similar resources.
+	resource.ImportStatePassthroughID(ctx, path.Root("app_endpoint_name"), req, resp)
+}
+
+// Create sets the default OIDC provider for the App Endpoint.
+func (r *AppEndpointDefaultOidcProvider) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.AppEndpointDefaultOidcProvider
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	organizationId := plan.OrganizationId.ValueString()
+	projectId := plan.ProjectId.ValueString()
+	clusterId := plan.ClusterId.ValueString()
+	appServiceId := plan.AppServiceId.ValueString()
+	appEndpointName := plan.AppEndpointName.ValueString()
+	providerId := plan.ProviderId.ValueString()
+
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s/oidcProviders/defaultProvider",
+		r.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+		appServiceId,
+		appEndpointName,
+	)
+
+	payload := api.AppEndpointOIDCDefaultProviderRequest{ProviderID: providerId}
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPut, SuccessStatus: http.StatusNoContent}
+	_, err := r.Client.ExecuteWithRetry(ctx, cfg, payload, r.Token, map[string]string{"Content-Type": "application/json"})
+	if err != nil {
+		resp.Diagnostics.AddError("Error Setting Default OIDC Provider", api.ParseError(err))
+		return
+	}
+
+	// Refresh state by listing providers and picking the default
+	selected, err := r.waitForDefaultProvider(ctx, organizationId, projectId, clusterId, appServiceId, appEndpointName, providerId)
+	if err != nil {
+		resp.Diagnostics.AddWarning("Error reading default OIDC provider after create", api.ParseError(err))
+	} else if selected.ProviderID != "" {
+		plan.ProviderId = types.StringValue(selected.ProviderID)
+	}
```

Comment:
is this an async API ?

there is no need to refresh as here as there are no computed attributes

Reply:
There is a slight delay in the default being set, I had an issue refreshing immediately after the PUT so i added a timer. Shouldnt there be a refresh to check that the operation succeeded?